tracer/src/Datadog.Trace/DataStreamsMonitoring/SpanExtensions.cs

```diff
@@ -25,10 +25,13 @@ internal static void SetDataStreamsCheckpoint(this Span span, DataStreamsManager
         }
 
         span.Context.SetCheckpoint(manager, checkpointKind, edgeTags, payloadSizeBytes, timeInQueueMs, parent);
-        var hash = span.Context.PathwayContext?.Hash.Value ?? 0;
-        if (hash != 0)
+        if (!manager.IsInDefaultState)
         {
-            span.SetTag("pathway.hash", hash.ToString());
+            var hash = span.Context.PathwayContext?.Hash.Value ?? 0;
+            if (hash != 0)
+            {
+                span.SetTag("pathway.hash", hash.ToString());
```

Comment:
Not related to the changes in this PR, and probably not required for `Int64`, but I always recommend specifying the culture when converting numbers to strings, so that we don't forget to use it when it _does_ matter:
```csharp
- span.SetTag("pathway.hash", hash.ToString());
+ span.SetTag("pathway.hash", hash.ToString(CultureInfo.InvariantCulture));
```